nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -1450,12 +1478,28 @@ def forward(
                     inference_max_sequence_len=inference_max_sequence_len,
                 )
             else:
```

Comment:
Same as above

Reply:
Discussed an approach with @titu1994 offline which I'll be implementing in a separate PR.